.drone.starlark

```diff
@@ -0,0 +1,950 @@
+			if 'emailNeeded' in config['defaults']['acceptance']:
+				default['emailNeeded'] = config['defaults']['acceptance']['emailNeeded']
+			if 'extraSetup' in config['defaults']['acceptance']:
+				default['extraSetup'] = config['defaults']['acceptance']['extraSetup']
+			if 'extraServices' in config['defaults']['acceptance']:
+				default['extraServices'] = config['defaults']['acceptance']['extraServices']
+			if 'extraApps' in config['defaults']['acceptance']:
+				default['extraApps'] = config['defaults']['acceptance']['extraApps']
+
+	for category, matrix in config['acceptance'].items():
+		if type(matrix['suites']) == "list":
+			suites = {}
+			for suite in matrix['suites']:
+				suites[suite] = suite
+		else:
+			suites = matrix['suites']
+
+		for suite, shortSuiteName in suites.items():
+			isWebUI = suite.startswith('webUI')
+			isAPI = suite.startswith('api')
+			isCLI = suite.startswith('cli')
+
+			if isAPI or isCLI:
+				default['browsers'] = ['']
+
+			servers = matrix['servers'] if 'servers' in matrix else default['servers']
+			browsers = matrix['browsers'] if 'browsers' in matrix else default['browsers']
+			phpVersions = matrix['phpVersions'] if 'phpVersions' in matrix else default['phpVersions']
+			databases = matrix['databases'] if 'databases' in matrix else default['databases']
+			federatedServerNeeded = matrix['federatedServerNeeded'] if 'federatedServerNeeded' in matrix else default['federatedServerNeeded']
+			filterTags = matrix['filterTags'] if 'filterTags' in matrix else default['filterTags']
+			emailNeeded = matrix['emailNeeded'] if 'emailNeeded' in matrix else default['emailNeeded']
+			extraSetup = matrix['extraSetup'] if 'extraSetup' in matrix else default['extraSetup']
+			extraServices = matrix['extraServices'] if 'extraServices' in matrix else default['extraServices']
+			extraApps = matrix['extraApps'] if 'extraApps' in matrix else default['extraApps']
+
+			for server in servers:
+				for browser in browsers:
+					for phpVersion in phpVersions:
+						for db in databases:
+							name = 'unknown'
+
+							if isWebUI or isAPI or isCLI:
+								browserString = '' if browser == '' else '-' + browser
+								name = '%s-%s%s-%s-php%s' % (shortSuiteName, server.replace('daily-', '').replace('-qa', ''), browserString, db.replace(':', ''), phpVersion)
+								maxLength = 50
+								nameLength = len(name)
+								if nameLength > maxLength:
+									print("Error: generated stage name of length", nameLength, "is not supported. The maximum length is " + str(maxLength) + ".", name)
+									errorFound = True
+
+							if isWebUI:
+								environment = {
+									'TEST_SERVER_URL': 'http://server',
+									'BEHAT_FILTER_TAGS': filterTags,
+									'BEHAT_SUITE': suite,
+									'SELENIUM_HOST': 'selenium',
+									'SELENIUM_PORT': '4444',
+									'BROWSER': browser,
+									'PLATFORM': 'Linux',
+								}
+
+							if isAPI or isCLI:
```

Comment:
set these settings in any case outside of the if, and remove them from `if isWebUI:`

Reply:
Done